pkglistgen.py

```diff
@@ -1135,6 +1135,12 @@ def do_update_and_solve(self, subcmd, opts):
 
         if not opts.project:
             raise ValueError('project is required')
+        opts.staging_project = None
+        if opts.scope.startswith('staging:'):
```

Comment:
how about not doing this here but instead use a few lines further down
if opts.scope.startswith("staging"), then try the match and set letters accordingly?

Reply:
the problem is the line in between those you commented - staging:X is not in SCOPES